sql/ddl_internal.sql

```diff
@@ -465,7 +465,36 @@ BEGIN
 END
 $BODY$;
 
+CREATE OR REPLACE FUNCTION _timescaledb_internal.verify_time_column_constraint(
+    main_table               REGCLASS,
+    column_name              NAME
+)
+    RETURNS VOID LANGUAGE PLPGSQL VOLATILE AS
+$BODY$
+DECLARE
+    is_not_null BOOLEAN;
+BEGIN
+    BEGIN
+        SELECT attnotnull
+        INTO STRICT is_not_null
+        FROM pg_attribute
+        WHERE attrelid = main_table AND attname = column_name;
+    EXCEPTION
+        WHEN NO_DATA_FOUND THEN
+            RAISE EXCEPTION 'column "%" does not exist', column_name
+            USING ERRCODE = 'IO102';
+    END;
+
+    -- The proper constraint is already set.
+    IF is_not_null THEN
+       RETURN;
+    END IF;
+
+    RAISE NOTICE 'Adding NOT NULL constraint to column %', column_name;
```

Comment:
FWIW, I would not raise this notice, because I think it can cause confusion as people might think they have done something wrong. We'll see how this plays out I guess, and how many support questions we get about this.

Reply:
I have no strong opinion but I thought that was decided from previous discussion ? @cevian @RobAtticus

Reply:
@olofr I think it was @cevian who wanted this. I mostly expressed my concern here, but figured it was already decided.

Reply:
I am in favor of this change. It's always nice for the user to be informed and it is not a warning not a blocking error, it seems like exactly the kind of thing for Notice. It could be considered "wrong" in that we don't allow it, so they know for the future not to do that, but either way I think it is a good thing to have.

Perhaps a lengthier message, e.g., `Adding NOT NULL constraint to time column % (NULL time values not allowed)`, is more helpful but that's a taste thing.